ethereum/spec/src/main/java/tech/pegasys/teku/spec/schemas/genesis/SchemaDefinitionsGenesis.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.spec.schemas.genesis;
+
+import java.util.List;
+import tech.pegasys.teku.datastructures.blocks.BeaconBlockHeader;
+import tech.pegasys.teku.datastructures.blocks.Eth1Data;
+import tech.pegasys.teku.datastructures.state.Checkpoint;
+import tech.pegasys.teku.datastructures.state.Fork;
+import tech.pegasys.teku.datastructures.state.PendingAttestation;
+import tech.pegasys.teku.datastructures.state.Validator;
+import tech.pegasys.teku.spec.constants.SpecConstants;
+import tech.pegasys.teku.spec.datastructures.state.BeaconStateFields;
+import tech.pegasys.teku.spec.datastructures.state.BeaconStateSchema;
+import tech.pegasys.teku.spec.datastructures.state.genesis.BeaconStateSchemaGenesis;
+import tech.pegasys.teku.spec.schemas.SchemaDefinitions;
+import tech.pegasys.teku.ssz.backing.schema.SszComplexSchemas;
+import tech.pegasys.teku.ssz.backing.schema.SszListSchema;
+import tech.pegasys.teku.ssz.backing.schema.SszPrimitiveSchemas;
+import tech.pegasys.teku.ssz.backing.schema.SszSchemaHints;
+import tech.pegasys.teku.ssz.backing.schema.SszVectorSchema;
+import tech.pegasys.teku.ssz.sos.SszField;
+import tech.pegasys.teku.util.config.Constants;
+
+public class SchemaDefinitionsGenesis implements SchemaDefinitions {
+  private final SpecConstants specConstants;
+
+  public SchemaDefinitionsGenesis(final SpecConstants specConstants) {
+    this.specConstants = specConstants;
+  }
+
+  @Override
+  public BeaconStateSchema<?> getBeaconStateSchema() {
+    // TODO - use specConstants throughout rather than deprecated CONSTANTS
+    // TODO - add methods to retrieve other schemas
+    SszField genesis_time_field =
+        new SszField(0, BeaconStateFields.GENESIS_TIME.name(), SszPrimitiveSchemas.UINT64_SCHEMA);
+    SszField genesis_validators_root_field =
+        new SszField(
+            1,
+            BeaconStateFields.GENESIS_VALIDATORS_ROOT.name(),
+            SszPrimitiveSchemas.BYTES32_SCHEMA);
+    SszField slot_field =
+        new SszField(2, BeaconStateFields.SLOT.name(), SszPrimitiveSchemas.UINT64_SCHEMA);
+    SszField fork_field = new SszField(3, BeaconStateFields.FORK.name(), Fork.SSZ_SCHEMA);
+    SszField latest_block_header_field =
+        new SszField(4, BeaconStateFields.LATEST_BLOCK_HEADER.name(), BeaconBlockHeader.SSZ_SCHEMA);
```

Comment:
Probably could look more natural if create fields inside `BeaconStateSchemaGenesis` with passing a `SpecConstants` instance there?

Reply:
May be also makes sense to share common field definitions ?

Reply:
Good idea - this area is definitely kind of messy.  Will update.